plugins/storage/volume/storpool/src/main/java/org/apache/cloudstack/storage/datastore/util/StorPoolUtil.java

```diff
@@ -240,6 +246,13 @@ public SpConnectionDesc(String url) {
             }
         }
 
+        private void extractUriParams(String url) throws URISyntaxException {
+            URI uri = new URI(url);
```

Comment:
@slavkap _storpool_  uri scheme validation required here?

_uri.getScheme() == null || !uri.getScheme().equalsIgnoreCase("storpool")_

Reply:
Thanks, @sureshanaparti! We don't use the scheme, that's why I don't check for it

Reply:
@slavkap new format (_storpool://{SP_AUTH_TOKEN}@{SP_API_HTTP}:{SP_API_HTTP_PORT}/{SP_TEMPLATE}_) has the scheme, so assuming it would be mentioned while creating StorPool primary storage. So, any scheme is accepted now, eg.  _nfs://, powerflex://, xyz://_ as it is ignored.

Reply:
@sureshanaparti, I've added the check.